### PR TITLE
[Network] Updated API version to 2017-06-01 and AvaialbleSslOptions Operation renamed

### PR DESCRIPTION
**Pull request by akshaysngupta**
Updated API version for Services contained in Network to 2017-06-01
AvaialbleSslOptions Operation renamed
…

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: arm-network/2017-06-01/swagger/applicationGateway.json
Comment (DeepakRajendranMsft) [inline arm-network/2017-06-01/swagger/applicationGateway.json]: this will be a breaking change

**Review by DeepakRajendranMsft — APPROVED** [review]

File: arm-network/2017-06-01/swagger/applicationGateway.json
Comment (akshaysngupta) [inline arm-network/2017-06-01/swagger/applicationGateway.json]: reverted.

File: arm-network/2017-06-01/swagger/applicationGateway.json
Comment (salameer) [inline arm-network/2017-06-01/swagger/applicationGateway.json]: This seems to return a list of Sub resource, am I correct? if so I think the right naming was chosen, thoughts @veronicagg & @balajikris

**Review by salameer — COMMENTED** [review]
whats the reason behind the name change?

File: arm-network/2017-06-01/swagger/applicationGateway.json
Comment (akshaysngupta) [inline arm-network/2017-06-01/swagger/applicationGateway.json]: AvailableSslOptions is new operation supported for application gateway which added in my previous PR (4c5dd5ded1f4a8cc26f7be35ad639c3e7380644d). I am undoing the name change as I might risk breaking something now.

**Comment by balajikris** [comment]
@akshaysngupta -- a nit, fix title and description please, as you're no longer changing any operationId, but only the API version

File: arm-network/2017-06-01/swagger/applicationGateway.json
Comment (balajikris) [inline arm-network/2017-06-01/swagger/applicationGateway.json]: From the internal email thread with Laurent who maintains Py SDK, it was implied that backward compat with 2017-03-01 was indeed a good thing, if not a requirement. With that in mind and the existing state of the spec, plus that this change adds no more errors/warns, I'm okay with keeping the existing bad naming for now.

**Comment by balajikris** [comment]
@akshaysngupta -- one more question. I see you are moving forward a bunch of specs to a newer version. As I understand these services may be split among different teams inside Networking. Have you gotten the okay from all the teams involved (whose specs were touched). ?

**Review by balajikris — CHANGES_REQUESTED** [review]
waiting on the author to respond to questions posed above.

**Comment by akshaysngupta** [comment]
Deepak is from Network RP team and he has approved the pull request for the version update for all the services. (Versions for all services are updated together to newer version).
Fixed the tiltle and description.

**Review by balajikris — APPROVED** [review]

**Review by balajikris — CHANGES_REQUESTED** [review]
@akshaysngupta : 
1. can you take a look at [this build output](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/246390805) and fix the problems indicated. This change broke model validation - `'No enum match for: 2017-06-01',`

File: arm-network/2017-06-01/swagger/applicationGateway.json
Comment (veronicagg) [inline arm-network/2017-06-01/swagger/applicationGateway.json]: I don't see AvailableSslOptions operation in 2017-03-01, it it's being introduced in this API, it'd be better to follow the naming convention. Did this operation exist before? @akshaysngupta what may break if this operation is named as you originally proposed in this PR?

Comment (akshaysngupta) [inline arm-network/2017-06-01/swagger/applicationGateway.json]: I am sure if I might break any dot net SDK PR

Comment (veronicagg) [inline arm-network/2017-06-01/swagger/applicationGateway.json]: @akshaysngupta can you link to it? If the SDK hasn't been generated/released (I assume it hasn't since this spec is new and under review), then you'd need to generate from the last version of the spec anyway (after merging this PR) and would need to re-run your tests to make sure things are working. Anyway, @balajikris can help evaluate, I'm just trying to provide information, so you can make a conscious decision.

**Comment by balajikris** [comment]
@akshaysngupta and I spoke offline and we agreed on a new set of changes that make sense. These changes have been pushed, waiting on CI

**Comment by balajikris** [comment]
model validation is failing due to a bug in the tool.  see https://github.com/Azure/oav/issues/132 .

**Review by balajikris — APPROVED** [review]

**Pull request merged**
